x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_management/logic/import/calculate_rule_source_for_import.test.ts

```diff
@@ -31,12 +32,58 @@ describe('calculateRuleSourceForImport', () => {
     });
   });
 
-  it('calculates as modified external type if an asset is found without a matching version', () => {
+  it('calculates as not modified external type if an asset is found without a matching version and no current rule present', () => {
     const rule = getRulesSchemaMock();
     rule.rule_id = 'rule_id';
 
     const result = calculateRuleSourceForImport({
-      rule,
+      importedRule: rule,
+      currentRule: undefined,
+      prebuiltRuleAssetsByRuleId: {},
+      isKnownPrebuiltRule: true,
+      ruleCustomizationStatus,
+    });
+
+    expect(result).toEqual({
+      ruleSource: {
+        type: 'external',
+        is_customized: false,
+      },
+      immutable: true,
+    });
+  });
+
+  it('calculates as non modified external type if an asset is found without a matching version and current rule present without changes', () => {
```

Comment:
There is a missing scenario covering that `is_customized: true` is preserved when imported rule matches with the current.